dashboard/app/helpers/jwt_verifier.rb

```diff
@@ -0,0 +1,51 @@
+class JwtVerifier
+  attr_reader :errors
+
+  def initialize(jwt, integration)
+    @jwt = jwt
+    @integration = integration
+    @errors = []
+  end
+
+  def verify_jwt
+    verify_audience(@jwt)
+    verify_expiration(@jwt)
+    errors.empty?
+  end
+
+  private
+
+  def verify_audience(jwt)
+    if jwt.key? :aud
+      aud = jwt[:aud]
+      aud_array = [*aud]
+      errors << 'Audience must be a string or Array of strings.' unless aud_array.all?(String)
+      if jwt.key? :azp
+        verify_azp(aud, jwt[:azp])
+      end
+    else
+      errors << 'Audience does not exist'
+    end
+  end
+
+  def verify_azp(aud, azp)
+    errors << 'Audience does not contain/match Authorized Party' unless azp_in_aud(aud, azp)
+  end
+
+  def verify_expiration(jwt)
```

Comment:
I wonder if we want to verify the issued at (iat)? Essentially make sure that this was issued at a time in the past, not future.

Reply:
Thoughts on doing that check as a followup? It doesn't seem blocking for the core functionality (based on the spec doc and jira ticket description).

Reply:
I'm fine doing it in a followup, I agree it's definitely not blocking anything. More of a (probably) best practice when validating JWTs. Want to add a jira task to track it?

Reply:
Jira task for tracking - https://codedotorg.atlassian.net/browse/P20-279

Will work on that next!